pxtlib/jsoutput.ts

```diff
@@ -284,6 +284,15 @@ namespace pxt.blocks {
         let output = ""
         let indent = ""
         let variables: Map<string>[] = [{}];
+        let currentLine = 0;
+
+        function append(s: string) {
+            // At runtime sometimes `s` is a number.
+            if (s?.match) {
```

Comment:
maybe `typeof(s) === "string"` ?

Reply:
I think I will go with that. It is more accurate in this case, and reinforces the comment above it.